ddspipe_core/src/cpp/core/DdsPipe.cpp

```diff
@@ -426,13 +429,12 @@ void DdsPipe::discovered_topic_nts_(
         // If Pipe is enabled and topic allowed, activate it
         if (enabled_ && allowed_topics_->is_topic_allowed(*topic))
         {
-            activate_topic_nts_(topic);
+            activate_topic_nts_(topic, endpoint_kind);
         }
     }
     else if (configuration_.remove_unused_entities && topic->topic_discoverer() != DEFAULT_PARTICIPANT_ID)
     {
-        // The bridge already exists. Create a writer in the participant who discovered it.
-        it_bridge->second->create_writer(topic->topic_discoverer());
+        it_bridge->second->create_endpoint(topic->topic_discoverer(), endpoint_kind);
```

Comment:
Design comment: call to endpoint creation method is done both from pipe and bridge. Analyze if it would be possible to further isolate their functionality so it's only done from one of them.

Reply:
Related: scope of `remove_unused_endpoint` (used both in pipe and bridge).